---
title: "Gene Analysis of Schizophrenia Patient Samples"
author: "Rui Xiang Yu & Nairuz Elazzabi (Pavlidis Lab)"
format: 
    html:
        toc: true
        toc-depth: 3
        auto-stretch: false
    pdf:
        toc: true
        toc-depth: 3 
        fig-pos: "H"
        auto-stretch: false
editor: source
bibliography: references.bib
execute:
    echo: false
    warning: false
---

```{r}
library(tidyverse)
library(knitr)
```

## Summary

TBF

## Introduction

TBF

## Methods

TBF

### Preview of the Data

The data used is from the @388Brains set of cohorts, with its samples. The raw data of the annotated expression matrices can be found [here](https://brainscope.gersteinlab.org/output-sample-annotated-matrix.html). All the samples come from the prefrontal cortex (PFC). The metadata contains the information of the following 12 cohorts:

1. CMC: used 10x Genomics Chip B with 10x Genomics protocol. Used Illumina Novaseq platform [@Ruzicka]
2. DevBrain
3. Girgenti-multiome
4. IsoHuB
5. LIBD
6. Ma_et_al [@MaSestan]
7. MultiomeBrain
8. PTSDBrainomics
9. SZBDMulti-Seq: used 10x Genomics Chip B. Libraries prepared with 10x Genomics Chromium Single Cell 3â€™ Reagent Kits v3 and MULTI-Seq protocols. Used Illumina NextSeq500, and an additional round with NovaSeq S2 [@Ruzicka]
10. ROSMAP
11. UCLA-ASD
12. Velmeshev_et_al [@Velmeshev]

However, because no annotated matrices were provided for the ROSMAP cohort, these were filtered out from the metadata.

This is an overview of each cohort:

```{r}
#| label: tbl-total-patients
#| tbl-cap: Each cohort with the total patients studied, the mean of the age of death, and disease studied.

total_patients <- readRDS("../results/1-total_patients.rds")
knitr::kable(total_patients)
```

From @tbl-total-patients, we can see that the CMC cohort has the biggest number of patient samples with 100 samples, followed by SZBDMulti-seq with 72. Ma_et_al has the smallest size, with only 2 samples. CMC and SZBDMulti-seq also have the highest mean of age of death at 71 years and 65 years respectively. Velmeshev_et_al has the lowest one, at 18 years. A variety of diseases are studied in these cohorts, with some cohorts studying none. These diseases include schizophrenia, autism spectrum disorder (ASD), bipolar disorder, major depressive disorder (MDD), post-traumatic stress disorder (PTSD), and Williams Syndrome.

```{r}
#| label: tbl-patientpercondition
#| tbl-cap: Number of patients per condition per cohort. Includes the mean age and the biological sex as well.

patientpercondition <- readRDS("../results/2-patients_per_condition.rds")
knitr::kable(patientpercondition)
```

@tbl-patientpercondition indicates the number of patients per condition in each cohort. The CMC cohort has the highest numbers, with 53 control samples and 47 schizophrenia samples. The UCLA-ASD is second, with 27 ASD samples and 25 control samples. The mean of age at death and the distribution of the biological sexes is also shown in this table.

It must be noted that many patients' age at death was marked as "89+". In order to determine the age distribution and the mean age of death, these values were converted to "89". The number of samples that were marked as "89+" from the Schizophrenia cohorts is in @tbl-elderlypatients.

```{r}
#| label: tbl-elderlypatients
#| tbl-cap: Number of patients annotated as "89+" in Schizophrenia cohorts.

elderlypatients <- readRDS("../results/3-elderlypatients.rds")
knitr::kable(elderlypatients)
```


The age at death distribution of each condition per cohort was also examined. In @fig-agedistribution, the dotted vertical lines are the mean of the age at death for each condition. From top to bottom and left to right:

- CMC: both conditions fall in the older range.
- IsoHub: there are only control samples and most fall in the younger range.
- MultiomeBrain: although most control samples fall between the 30-45 range, there are samples that are younger than 25 and older than 50. Schizophrenia samples fall around the 50 year age range, and BPD samples fall between the 30-45 range.
- UCLA-ASD: both control and ASD samples fall in the younger range, with an average of around 25 years.
- DevBrain: control samples fall between the 25-50 range. ASD samples faill in the younger range, with an average of around 25 years. Williams Syndrome samples are also in the younger range, although their average is slightly higher than 25.
- LIBD: there are only control samples and most of them fall in the 50 year range.
- PTSDBrainomics: control and MDD samples fall in the 50 year range. PTSD samples are slightly younger in average, but their distribution is wide.
- Velmeshev_et_al: most ASD and control samples are under 25 years old.
- Girgenti-snMultiome: there are only control samples. The average is around 50 years old.
- Ma-Sestan: there are only control samples. There is one sample that falls around 50 years old, and the other that falls around 60 years old.
- SZBDMulti-Seq: both control and Schizophrenia samples have similar distributions ad averages of around 60 years. THe BPD samples' distribution is shifted slightly more towards the right, with an older average of around 60 years old.

![Age distribution in each cohort.](../results/4-agedistribution.png){#fig-agedistribution}

### Examining the Schizophrenia Cohorts

The number of genes and single cells per each Schizophrenia cohort was also looked at in @tbl-genescells. The number of genes is per sample, whereas the number of cells is the total number for all samples in a cohort. 

```{r}
#| label: tbl-genescells
#| tbl-cap: Number of genes and total number of cells per each Schizophrenia cohort.

cells_genes <- readRDS("../results/5-cells_genes.rds")
knitr::kable(cells_genes)
```

The number of genes is quite similar across 3 cohorts. SZBDMulti-Seq has the highest number of total cells, at 603281. MultiomeBrain has the lowest number of total cells, at 134666.

From @Ruzicka, genes expressed in less than 0.1% cells and cells with fewer than 1000 identified genes were removed. If they had more than 10% unique molecular identifiers in mitochondrial genes or more than 50000 captured UMIs, these were also removed. They also determined outlier cells by checking their total UMI count, percentage of reads mapped to mitochondrial genes, low similarity with other cells, and exploration of 2D projection plots. These were also removed.

From @ling2024concerted, donors with less than 1000 UMIS were excluded. For each cell type, matrices were scaled to 100000 UMIs per donor and filtered to the top expressing genes. Top expressing genes had at least 10 UMIs per 100000 for at least one donor.

For our purposes, genes that are not expressed in at least 5% of cells will be filtered out. Cells must also be in the top 5th percentile in terms of the number of expressed genes. If the cell comes from a patient that had less than 20 cells for that specific cell type, these will be filtered out. The single cells will be CPM and log normalized. The cell types will be grouped into major types:

- Excitatory neurons: these include neurons that are L2/3 intratelencephalic, L4 IT intratelencephalic, L5 intratelencephalic, L6 intratelencephalic, L6 corticothalamic, L6 intratelencephalic Car3, L5 extratelencephalic, LE/6 near-projecting, and L6b intratelencephalic.
- Inhibitory neurons: these chandelier cells and neurons that express somatostatin, somatostatin and chondrolectin, parvalbumin, paired box 6, lysosomal-associated membrane protein 5 and LIM homeobox protein 6, lysosomal-associated membrane protein 5, synuclein gamma, and vasoactive intestinal polypeptide.
- Astrocytes
- Microglia cells
- OLigodendrocyte progenitor cells
- Oligodendrocytes

In @tbl-genescellsfiltered, we can see that the number of genes still remains quite similar across three cohorts after filtering. Approximately 23000 genes were removed from each cohort. Not that many cells were removed from the CMC cohort. A moderate amount was removed from the SZBDMulti-Seq cohort, and a substantial amount (almost half) was removed from the MultiomeBrain cohort.

```{r}
#| label: tbl-genescellsfiltered
#| tbl-cap: Number of genes and total number of cells per each Schizophrenia cohort after filtering.

cells_genes_filtered <- readRDS("../results/5.2-cells_genes.rds")
knitr::kable(cells_genes_filtered)
```

The next thing that was looked into was the number of cells per cell-type per patient. In @fig-celltypedistribution, we can see that L2/3 intratelencephalic excitatory neurons have the highest frequency (279756 in total), especially by the SZBDMulti-Seq cohort. This is followed by oligodendrocytes with 221554 counts in total. Sst Chodl inhibitory neurons have the lowest count at 746 counts. Smooth muscle cells (SMC) have the second lowest count, at 944.

In @fig-depthdistribution, the sequencing depth of each cell was investigated. These results were grouped together per cell type in the following plot, and colored by cohort.

Lastly, the difference in cell type abundance per condition was investigated in each cohort in @fig-abundancecelltype. 

![Cells per cell-type per patient in each cohort.](../results/6-celltypedistribution.png){#fig-celltypedistribution}

![Depth distribution per cell-type per patient in each cohort.](../results/7-depthdistribution.png){#fig-depthdistribution}

![Cell type abundance within conditions across cohorts.](../results/8-abundancecondition.png){#fig-abundancecelltype}

## References
