---
title: "Gene Analysis of Schizophrenia Patient Samples"
author: "Rui Xiang Yu & Nairuz Elazzabi (Pavlidis Lab)"
number-sections: true
format: 
    html:
        toc: true
        toc-depth: 3
        auto-stretch: false
        embed-resources: true
    pdf:
        toc: true
        toc-depth: 3 
        fig-pos: "H"
        auto-stretch: false
editor: source
bibliography: references.bib
execute:
    echo: false
    warning: false
---

```{r}
library(tidyverse)
library(knitr)
```

## Summary

TBF

## Introduction

Schizophrenia  (SCZ) is a mental disorder that affects 0.3-0.7%. It has many symptoms, including delusions and hallucinations. Although many studies have come out studying the transcriptome of patients with this disorder, a clearer picture is needed. This project aims to uncover the transcriptome of Schizophrenia patients, which cell types are associated with this disorder, and review past methods from other papers.

## Methods

The following sub-segments cover the methods employed, as well as the results obtained.

### Preview of the Data

The data used is from the @388Brains set of cohorts, with its samples. The raw data of the annotated expression matrices can be found [here](https://brainscope.gersteinlab.org/output-sample-annotated-matrix.html). The raw metadata can be found [here](https://brainscope.gersteinlab.org/integrative_files.html) under the name of PEC2_sample_mapping.xlsx. 

An additional cohort, named Batiuk, was included in the study as well [@batiuk]. Its matrices and cell-type annotations can be found [here](https://zenodo.org/records/6921620). Its metadata can be found as a supplementary file in its paper.

All the samples come from the prefrontal cortex (PFC). These are all the cohorts:

1.  CMC [@Ruzicka, @cmc2]
2.  DevBrain
3.  Girgenti-multiome
4.  IsoHuB
5.  LIBD [@huuki, @yap]
6.  Ma_et_al [@MaSestan]
7.  MultiomeBrain [@multiomebrain]
8.  PTSDBrainomics [@ptsdbrainomics]
9.  SZBDMulti-Seq [@Ruzicka]
10. ROSMAP
11. UCLA-ASD [@ucla-asd1, @ucla-asd2]
12. Velmeshev_et_al [@Velmeshev]
13. Batiuk [@batiuk]

However, because no annotated matrices were provided for the ROSMAP cohort, these were filtered out from the metadata.

This is an overview of each cohort:

```{r}
#| label: tbl-total-patients
#| tbl-cap: Each cohort with the total patients studied, the mean of the age of death, and disease studied.

total_patients <- readRDS("../results/1-total_patients.rds")
knitr::kable(total_patients)
```

From @tbl-total-patients, we can see that the CMC cohort has the biggest number of patient samples with 100 samples, followed by SZBDMulti-seq with 72. Ma_et_al has the smallest size, with only 2 samples. CMC has the highest mean in age, at 71 years old. SZBDMulti-Seq and Batiuk share the same mean age, at 65 years old. Velmeshev_et_al has the lowest one, at 18 years. A variety of diseases are studied in these cohorts, with some cohorts studying none. These diseases include schizophrenia, autism spectrum disorder (ASD), bipolar disorder, major depressive disorder (MDD), post-traumatic stress disorder (PTSD), and Williams Syndrome.

```{r}
#| label: tbl-patientpercondition
#| tbl-cap: Number of patients per condition per cohort. Includes the mean age and the biological sex as well.

patientpercondition <- readRDS("../results/2-patients_per_condition.rds")
knitr::kable(patientpercondition)
```

@tbl-patientpercondition indicates the number of patients per condition in each cohort. The CMC cohort has the highest numbers, with 53 control samples and 47 schizophrenia samples. The UCLA-ASD is second, with 27 ASD samples and 25 control samples. The mean of age at death and the distribution of the biological sexes is also shown in this table.

It must be noted that many patients' age at death was marked as "89+". In order to determine the age distribution and the mean age of death, these values were converted to "89". The number of samples that were marked as "89+" from the Schizophrenia cohorts is in @tbl-elderlypatients.

```{r}
#| label: tbl-elderlypatients
#| tbl-cap: Number of patients annotated as "89+" in Schizophrenia cohorts.

elderlypatients <- readRDS("../results/3-elderlypatients.rds")
knitr::kable(elderlypatients)
```

The age at death distribution of each condition per cohort was also examined. In @fig-agedistribution, the dotted vertical lines are the mean of the age at death for each condition. From top to bottom and left to right:

-   CMC: both conditions fall in the older range.
-   DevBrain: control samples fall between the 25-50 range. ASD samples faill in the younger range, with an average of around 25 years. Williams Syndrome samples are also in the younger range, although their average is slightly higher than 25.
-   Girgenti-snMultiome: there are only control samples. The average is around 50 years old.
-   Batiuk: the age distribution of both groups appear similar, within the 50-75 age range. The average of the Schizophrenia group is slightly older.
-   IsoHub: there are only control samples and most fall in the younger range.
-   LIBD: there are only control samples and most of them fall in the 50 year range.
-   SZBDMulti-Seq: both control and Schizophrenia samples have similar distributions ad averages of around 60 years. THe BPD samples' distribution is shifted slightly more towards the right, with an older average of around 60 years old.
-   Ma-Sestan: there are only control samples. There is one sample that falls around 50 years old, and the other that falls around 60 years old.
-   PTSDBrainomics: control and MDD samples fall in the 50 year range. PTSD samples are slightly younger in average, but their distribution is wide.
-   MultiomeBrain: although most control samples fall between the 30-45 range, there are samples that are younger than 25 and older than 50. Schizophrenia samples fall around the 50 year age range, and BPD samples fall between the 30-45 range.
-   UCLA-ASD: both control and ASD samples fall in the younger range, with an average of around 25 years.
-   Velmeshev_et_al: most ASD and control samples are under 25 years old.

![Age distribution in each cohort.](../results/4-agedistribution.png){#fig-agedistribution}

### Examining the Schizophrenia Cohorts

The number of patients, split by disorder and colored by sex, can be seen in @fig-patientbarplot. 

![Bar plot of number of patients per cohort, split into disorder and colored by sex.](../results/12-patientbarplot.png){#fig-patientbarplot}

The distribution of ages was looked at in @fig-age_dist_plots. 

- Top left: histogram of the ages of all patients corresponding to the Schizophrenia cohorts. Most individuals are between the ages of 60 and 90, and a peak is observed at 89 years. 
- Top right: quantile-quantile (QQ) plot of the ages from Schizophrenia patients compared to those of the control. Since most points fall along the reference line, the age distributions between the two groups are quite similar. The deviation seen at the ends of the line suggests differences in the distribution tails, most likely by the presence of older individuals in the control group than in the diseased group.
- Middle left: histogram of the ages of all patients corresponding to the Schizophrenia cohorts, colored by disease. The Schizophrenia group has a lower median. We can also see that there are more older individuals from the control group. 
- Middle right: violin plot of the age distributions for the control and Schizophrenia groups. The shape and spread shows that the density and distribution of both groups are quite similar, with differences in the median.
- Bottom left: histogram of the ages of all patients corresponding to the Schizophrenia cohorts, colored by sex.  The MALE group has a lower median. We can also see that there are more older individuals from the female group. 
- Bottom right: violin plot of the age distributions for the female and male groups. The female group exhibits a broader and slightly higher age range than the male group.

The distribution of number of cells per patient was looked at in @fig-cellcount_dist_plots. 

- Top left: violin plot of the cell count distribution compared by condition. The CMC cohort's schizophrenia cell counts appear to be generally lower than the control group, with the control group having a broader range. The MultiomeBrain cohort has a narrow distribution in both groups, with the Schizophrenia group having a higher median and slightly wider spread. In the SZBDMulti-Seq cohort, the schizophrenia group has a broader distribution.
- Top right: QQ-plot of the cell counts between control and Schizophrenia groups. It indicates that there are similar distributions at lower cell counts, but deviations at higher quantiles suggest that one condition often has larger cell counts per patient.
- Bottom left: violin plot of the cell count distribution compared by sex. In the CMC cohort, the shape and spread of both groups appears to be quite similar. In the MultiomeBrain cohort, the female cohort has a higher median and broader spread than the male group. In the SZBDMulti-Seq group, the male group has a broader distribution.
- Bottom right: QQ-plot of the cell counts between female and male groups. There are similar distributions at lower cell counts, but deviations in higher quantiles suggest that one group often has higher cell counts in the upper range.

![Age distributions across cohorts, compared by disease state and biological sex.](../results/10-age_distributions.png){#fig-age_dist_plots}

![Cell counts per patient across cohorts, compared by disease state and biological sex.](../results/11-cellcount_plots.png){#fig-cellcount_dist_plots}

The number of genes and single cells per each Schizophrenia cohort was also looked at in @tbl-genescells. The number of genes is per sample, whereas the number of cells is the total number for all samples in a cohort.

```{r}
#| label: tbl-genescells
#| tbl-cap: Number of genes and total number of cells per each Schizophrenia cohort.

cells_genes <- readRDS("../results/5-cells_genes.rds")
knitr::kable(cells_genes)
```

The number of genes is quite similar across 3 cohorts. SZBDMulti-Seq has the highest number of total cells, at 603281. MultiomeBrain has the lowest number of total cells, at 134666.

From @Ruzicka, genes expressed in less than 0.1% cells and cells with fewer than 1000 identified genes were removed. If they had more than 10% unique molecular identifiers in mitochondrial genes or more than 50000 captured UMIs, these were also removed. They also determined outlier cells by checking their total UMI count, percentage of reads mapped to mitochondrial genes, low similarity with other cells, and exploration of 2D projection plots. These were also removed.

For our purposes, genes that are not expressed in at least 5% of cells will be filtered out. Cells must also be in the top 5th percentile in terms of the number of expressed genes. If the cell comes from a patient that had less than 20 cells for that specific cell type, these will be filtered out. The single cells will be CPM and log normalized. The cell types will be grouped into major types:

-   Excitatory neurons: these include neurons that are L2/3 intratelencephalic, L4 IT intratelencephalic, L5 intratelencephalic, L6 intratelencephalic, L6 corticothalamic, L6 intratelencephalic Car3, L5 extratelencephalic, LE/6 near-projecting, and L6b intratelencephalic.
-   Inhibitory neurons: these chandelier cells and neurons that express somatostatin, somatostatin and chondrolectin, parvalbumin, paired box 6, lysosomal-associated membrane protein 5 and LIM homeobox protein 6, lysosomal-associated membrane protein 5, synuclein gamma, and vasoactive intestinal polypeptide.
-   Astrocytes
-   Microglia cells
-   OLigodendrocyte progenitor cells
-   Oligodendrocytes

In @tbl-genescellsfiltered, we can see that the number of genes still remains quite similar across three cohorts after filtering. Approximately 23000 genes were removed from each cohort. Not that many cells were removed from the CMC cohort. A moderate amount was removed from the SZBDMulti-Seq cohort, and a substantial amount (almost half) was removed from the MultiomeBrain cohort.

```{r}
#| label: tbl-genescellsfiltered
#| tbl-cap: Number of genes and total number of cells per each Schizophrenia cohort after filtering.

cells_genes_filtered <- readRDS("../results/5.2-cells_genes.rds")
knitr::kable(cells_genes_filtered)
```

The next thing that was looked into was the number of cells per cell-type per patient. In @fig-celltypedistribution, we can see that excitatory neurons have the highest frequency, especially by the SZBDMulti-Seq cohort. This is followed by oligodendrocytes. Microglia and oligodendrocyte progenitor cells have the lowest frequencies.

The sequencing depth of each cell was investigated, grouped by cell type. @fig-1depthdistribution shows the sequencing depth distribution with all the x-axis at the same scale. @fig-2depthdistribution shows the sequencing depth distribution with all the x-axis unscaled. @fig-3depthdistribution shows the sequencing depth distribution after the matrices had been CPMLog normalized, with all the x-axis at the same scale. @fig-4depthdistribution shows the sequencing depth distrbution after CPMLog normalization, but with the x-axis unscaled. Across all three cohorts, excitatory neurons have the highest sequencing depth, especially in the MultiomeBrain cohort. Microglia have the lowest sequencing depth in all cohorts.

The difference in cell type abundance per condition was investigated in each cohort in @fig-abundancecelltype. It appears that control samples have a slightly higher number of cells than disease samples. In CMC's microglia, MultiomeBrain's astrocytes and oligodendroctes, SZBDMulti-Seq's astrocytes and OPCs, the disease samples have a slighly higher number of cells. In accordance with what has been seen so far, excitatory neurons have the highest frequency while microglia have the lowest.

![Cells per cell-type per patient in each cohort.](../results/6-celltypedistribution.png){#fig-celltypedistribution}

![Depth distribution per cell-type per patient in each cohort, scaled equally.](../results/7-depthdistribution.png){#fig-1depthdistribution}

![Depth distribution per cell-type per patient in each cohort, unscaled.](../results/7.2-depthdistribution_unscaled.png){#fig-2depthdistribution}

![Depth distribution per cell-type per patient in each cohort after CPMLog normalization, scaled equally.](../results/7.3-depthdistribution_cpm.png){#fig-3depthdistribution}

![Depth distribution per cell-type per patient in each cohort after CPMLog normalization, unscaled.](../results/7.4-depthdistribution_cpm_unscaled.png){#fig-4depthdistribution}

![Cell type abundance within conditions across cohorts.](../results/8-abundancecondition.png){#fig-abundancecelltype}

UMAP was performed for each cohort, colored according to different characteristics. In @fig-celltypeumap, we can see that the cell types were assigned to different clusters. The cluster profile of the CMC and SZBDMulti-Seq cohort appear to be quite similar. The cell type assignment for the MultiomeBrain cohort does appear to be slightly different, most likely due to the fact it is the smallest cohort. @fig-disorderumap shows the UMAP assignment by disorder and @fig-sexumap by biological sex. It appears that both conditions/characteristics are evenly spread across cells.

![Assignments of nuclei to cell types, separated by cohort.](../results/9.10-cell_type_umap.png){#fig-celltypeumap}

![Assignments of nuclei to disease state, separated by cohort.](../results/9.11-disorder_umap.png){#fig-disorderumap}

![Assignments of nuclei to biological sex, separated by cohort.](../results/9.12-sex_umap.png){#fig-sexumap}

## References
